GeosCore/diagnostics_mod.F90

```diff
@@ -1659,4 +1660,498 @@ SUBROUTINE Do_Archive_SatDiagn( Input_Opt,  State_Chm,  State_Diag,        &
+    REAL(fp),      POINTER :: SOAS        (:,:,:)
+    REAL(fp),      POINTER :: FRAC_SNA    (:,:,:,:)
+    REAL(fp),      POINTER :: DAERSL      (:,:,:,:)
+    REAL(fp),      POINTER :: WAERSL      (:,:,:,:)
+
+    ! Conversionf factors to ugC/m3 for Total Organic Carbon diagnostic
+    REAL(fp), SAVE :: Fac_INDIOL
+    REAL(fp), SAVE :: Fac_LVOCOA
+    REAL(fp), SAVE :: Fac_SOAGX
+    REAL(fp), SAVE :: Fac_SOAIE
+
+    ! Species ids
+    INTEGER,  SAVE :: id_INDIOL
+    INTEGER,  SAVE :: id_LVOCOA
+    INTEGER,  SAVE :: id_SOAGX
+    INTEGER,  SAVE :: id_SOAIE
+!
+! !DEFINED PARAMETERS:
+!
+    ! Convert [kg/m3] to [ug/m3]
+    REAL(fp),      PARAMETER :: kgm3_to_ugm3 = 1.0e+9_fp
+
+    ! Define number of carbon atoms in each irreversible isoprene
+    ! SOA tracer species. Named according to the parent HC (same
+    ! number of carbons):
+    REAL(fp),      PARAMETER :: NCIMAE   = 4e+0_fp
+    REAL(fp),      PARAMETER :: NCIEPOX  = 5e+0_fp
+    REAL(fp),      PARAMETER :: NCINDIOL = NCIEPOX
+    REAL(fp),      PARAMETER :: NCGLYX   = 2e+0_fp
+    REAL(fp),      PARAMETER :: NCGLYC   = NCGLYX
+    REAL(fp),      PARAMETER :: NCMGLY   = 3e+0_fp
+    REAL(fp),      PARAMETER :: NCLVOC   = NCIEPOX
+    REAL(fp),      PARAMETER :: NCISN1   = NCIEPOX
+
+    !=======================================================================
+    ! Set_AerMass_Diagnostic begins here!
+    !=======================================================================
+
+    ! Initialize
+    RC       = GC_SUCCESS
+    ErrMsg   = ''
+    ThisLoc  = ' -> at Set_AerMass_Diagnostic (in module GeosCore/aerosol_mod.F90)'
+
+    ! Set pointers
+    OCFPOA      => State_Chm%AerMass%OCFPOA
+    OCFOPOA     => State_Chm%AerMass%OCFOPOA
+    BCPI        => State_Chm%AerMass%BCPI
+    BCPO        => State_Chm%AerMass%BCPO
+    OCPI        => State_Chm%AerMass%OCPI
+    OCPO        => State_Chm%AerMass%OCPO
+    OCPISOA     => State_Chm%AerMass%OCPISOA
+    SALA        => State_Chm%AerMass%SALA
+    ACL         => State_Chm%AerMass%ACL
+    SALC        => State_Chm%AerMass%SALC
+    SO4_NH4_NIT => State_Chm%AerMass%SO4_NH4_NIT
+    SO4         => State_Chm%AerMass%SO4
+    HMS         => State_Chm%AerMass%HMS
+    NH4         => State_Chm%AerMass%NH4
+    NIT         => State_Chm%AerMass%NIT
+    SLA         => State_Chm%AerMass%SLA
+    SPA         => State_Chm%AerMass%SPA
+    TSOA        => State_Chm%AerMass%TSOA
+    ASOA        => State_Chm%AerMass%ASOA
+    OPOA        => State_Chm%AerMass%OPOA
+    SOAGX       => State_Chm%AerMass%SOAGX
+    PM25        => State_Chm%AerMass%PM25
+    PM10        => State_Chm%AerMass%PM10
+    ISOAAQ      => State_Chm%AerMass%ISOAAQ
+    SOAS        => State_Chm%AerMass%SOAS
+    FRAC_SNA    => State_Chm%AerMass%FRAC_SNA
+    DAERSL      => State_Chm%AerMass%DAERSL
+    WAERSL      => State_Chm%AerMass%WAERSL
+
+    ! Check that species units are kg/kg dry air
+    IF ( TRIM( State_Chm%Spc_Units ) /= 'kg/kg dry' ) THEN
```

Comment:
This will cause a merge conflict when merging into dev/no-diff-to-benchmark.  Just to be aware.

Reply:
I rebased my branch onto the dev/no-diff-to-benchmark and made the updates needed compatibility.